bundle-generator/deployment/src/main/java/io/quarkiverse/operatorsdk/bundle/deployment/BundleProcessor.java

```diff
@@ -79,16 +81,17 @@ CSVMetadataBuildItem gatherCSVMetadata(CombinedIndexBuildItem combinedIndexBuild
     }
 
     @BuildStep
-    void generateCSV(CSVGenerationConfiguration configuration,
+    void generateBundle(BundleGenerationConfiguration configuration,
+            BuildTimeOperatorConfiguration operatorConfiguration,
             OutputTargetBuildItem outputTarget,
             CSVMetadataBuildItem csvMetadata,
-            BuildProducer<GeneratedCSVBuildItem> doneGeneratingCSV,
+            BuildProducer<GeneratedBundleBuildItem> doneGeneratingCSV,
             GeneratedCRDInfoBuildItem generatedCustomResourcesDefinitions,
             List<GeneratedKubernetesResourceBuildItem> generatedKubernetesManifests,
             BuildProducer<GeneratedFileSystemResourceBuildItem> generatedCSVs) {
-        if (configuration.generateCSV.orElse(false)) {
+        if (configuration.enabled) {
```

Comment:
Wondering if the default for this shouldn't be `false` as we don't want to be generating the bundle for every build… Maybe we should only activate the extension during a `package` build?

Reply:
It would be odd to me that we add the bundle dependency and that nothing different happens :/
What do you mean by enabling this extension only a `package` build? 
A final note: generating the bundle does not mean that we're going to deploy the operator, but users can use the generated bundle manifests to proceed with the process.

Reply:
Yes, but I'm trying to keep the build as light as possible, i.e. it shouldn't do things that are not needed. For example, the CRD is only generated if the extension detects that the classes impacting it have changed, otherwise the CRD generation is skipped. Same goes for the controller configuration. 

I agree that disabling the extension by default would be weird but it also doesn't make sense to generate the bundle on every build. It really only makes sense to generate it whenever we ask for the executable to be generate i.e. during a `mvn package` build. So, ideally, the bundle extension would only activate during `mvn package`.

Reply:
As discussed, we'll document the bundle extension to be used only when releasing operators. This will be documented in https://github.com/quarkiverse/quarkus-operator-sdk/issues/245